Pull request for review 2

EvgeniiyaR opened the pull request:
В данном Pull request находится актуальное состояние ветки Develop.

MaxOrel commented on src/components/App/App.jsx:
```diff
@@ -1,39 +1,183 @@
+			getUserInformation()
+				.then((user) => {
+					setIsLoggedIn(true);
+					if (user.role === 'volunteer') {
```
Лучше в таких случаях использовать switch case, а не if

MaxOrel commented on src/components/App/App.jsx:
```diff
@@ -1,39 +1,183 @@
-		first_name: '',
-		second_name: '',
-		last_name: '',
+	const [currentUser, setCurrentUser] = useState({
```
Не думали завести централизованное хранилище Redux или Mobx?

MaxOrel commented on src/components/App/App.jsx:
```diff
@@ -1,39 +1,183 @@
+		Promise.all([getNews(), getPlatformAbout()])
+			.then(([dataNews, dataPlatformAbout]) => {
+				setNews(dataNews.results);
+				const { about_us: aboutUs, valuations } = dataPlatformAbout;
```
Существует такой паттерн программирования называется адаптер, вот чтобы не деструктурировать с переименованием. Можете его применить кога вас не удовлетворяют поля которые приходят с сервера https://doka.guide/js/design-patterns-structural/

MaxOrel commented on src/components/App/App.jsx:
```diff
@@ -1,39 +1,183 @@
+		closeModal();
+		logOut()
+			.then(() => {
+				localStorage.removeItem('token');
```
Вынесите работу с localStorage в отдельные функции, где ключ вы будете передавать как аргумент.

MaxOrel commented on src/routes/router.js:
```diff
@@ -1,31 +1,87 @@
+import ProfileOrganizationEdit from '../components/ProfileOrganizationEdit/ProfileOrganizationEdit';
 
-const router = createBrowserRouter([
+const router = createHashRouter([
```
Почему используете hash router? Если хотите публиковать на githubPages можете завести переменную окружения и если деплой идет на github pages то подставлять hashrouter, а вообще лучше использовать brouserRouter

MaxOrel commented on src/components/App/App.jsx:
```diff
@@ -61,18 +243,38 @@ function App() {
 			<Footer platformEmail={platformEmail} />
-			<Modal modal={modal} closeModal={closeModal} />
+			{modal.isOpen &&
+				createPortal(
```
Портал нужно создать внутри самого компонента модального окна, т.е. обернуть его в createPortal, чтобы каждый раз не подключать его

MaxOrel reviewed (approved):
Работа в целом написана хорошо, лучше конечно завести централизованное хранилище, т.к. данные размазаны по всему проекту и следить за ними и контролировать становится все сложнее. В качестве хранилища используйте либо Mobx, либо Redux Toolkit

EvgeniiyaR committed:
fix(request): use .some

EvgeniiyaR committed:
feat(router): add route projects/:id

EvgeniiyaR committed:
fix(organizer): delete token

EvgeniiyaR committed:
feat(projectview): add ProjectView component

EvgeniiyaR committed:
feat(projects): add Link for CardProject

EvgeniiyaR committed:
feat(modal): add style for enlarge-image

EvgeniiyaR committed:
fix(cardnews): fix format date

EvgeniiyaR committed:
fix(cardnews): ul to div

DBKai committed:
fix(customtextarea): removed type prop

EvgeniiyaR committed:
feat(projectview): if idProject does not exist then Not Found component opens

EvgeniiyaR committed:
fix(signupapi): add return

EvgeniiyaR committed:
refactor(projectview): eslint-disable-next-line no-alert temporarily

EvgeniiyaR committed:
feat: resolve conflicts

EvgeniiyaR committed:
Merge pull request #169 from volunteers-for-city-projects/feature/project-view

Feature/project view

Julia-Papina committed:
feat: day and time on card project

EvgeniiyaR committed:
fix(modal): change style

EvgeniiyaR committed:
fix(request): change logic handle errors

DBKai committed:
feat(selectoption): changed error transmission method

DBKai committed:
feat(project): changed error transmission method

DBKai committed:
feat(project): added save values to localstorage

vvkon13 committed:
feat(projects): add a component for displaying the project status bar in project card

vvkon13 committed:
fix: lf crlf

Commit:
feat(projectview): add realise like and dislike project faeture  p7.6

fix #160

KPVakhrushev committed:
style(component): fix spelling errors

VanVorobyov committed:
fix(signup-forms): fix styles and add converter of quotation

vvkon13 committed:
fix(login): add Login 1366 px display

vvkon13 committed:
fix(login): lf crlf

VanVorobyov committed:
fix(add-project): fix select input

Commit:
feat(cardpriject projects projectview): implement adding a project to favorites

implement adding a project to favorites from projects page and projectView page and adding condition
isLoggedIn for display likebtn

#160

KPVakhrushev committed:
Merge branch 'develop' into feature/project-incomes

Julia-Papina committed:
feat: button 1366 on projects page

vvkon13 committed:
Merge pull request #173 from volunteers-for-city-projects/fix/login-1366px

Fix/login 1366px

vvkon13 committed:
fix(showprojectstatus): fix the prop type error in the props component of ShowProjectStatus

vvkon13 committed:
fix(showprojectstatus): change the prop type of the list to the line independence from backend

vvkon13 committed:
Merge pull request #171 from volunteers-for-city-projects/feature/projects-preview-function-status-project

Feature/projects preview function status project

Julia-Papina committed:
feat: change img to a button in card project, prop types button, screen 1366

Julia-Papina committed:
feat(fix conflicts from the develop): fix card news length

Julia-Papina committed:
feat: change styles card project

KPVakhrushev committed:
feat(component): income form

Julia-Papina committed:
style(height  card  project  1366): style

Julia-Papina committed:
fix: added overflow: hidden and text-overflow for text components in layout

VanVorobyov committed:
feat(projects-select-input): add mask to date range input

VanVorobyov committed:
Merge pull request #174 from volunteers-for-city-projects/fix/forms

Fix/forms

Commit:
Merge branch 'develop' into feature/p.7.6_add_project_to_favorite

Julia-Papina committed:
style(slider): blue color, centered

Julia-Papina committed:
feat: projects organization on personal profile

Commit:
fix(projects projectview organaizer, projectlikebutton): implement adding project for favorites

change logic of getting projects  array for logged and unlogged user, change size of likebtn for
diff pages

Julia-Papina committed:
feat(card project): block for buttons on card project

Julia-Papina committed:
Merge branch 'develop' into feature/projects-modification

Julia-Papina committed:
Merge pull request #177 from volunteers-for-city-projects/feature/projects-modification

Feature/projects modification

Julia-Papina committed:
fix(projects.scss): changed styles projects__selects

Julia-Papina committed:
Merge pull request #178 from volunteers-for-city-projects/fix/projects-option-selects

fix(projects.scss): changed styles projects__selects

Julia-Papina committed:
feat(merge develop): conflict resolution

Julia-Papina committed:
feat: added new button tabs in profile, changed layout new design, api project me

VanVorobyov committed:
feat(projects): add date picker for projects filter + add styles

VanVorobyov committed:
style(datepicker): add styles to header

KPVakhrushev committed:
Merge pull request #165 from volunteers-for-city-projects/feature/project-incomes

Feature/project incomes

KPVakhrushev committed:
feat(component): added component ModalContent

Julia-Papina committed:
style: styles profile volunteers and organization

KPVakhrushev committed:
Merge branch 'develop' into feature/project-incomes

Julia-Papina committed:
feat: project delete button, button tabs in profile volunteer, changed styles

Julia-Papina committed:
feat: new class name in profile organization, styles profile, added icon pensil in card project

VanVorobyov committed:
fix(projects): fix prop types in projects inputs

VanVorobyov committed:
Merge pull request #180 from volunteers-for-city-projects/fix/forms

Fix/forms

VanVorobyov committed:
feat(projects): add city filter

VanVorobyov committed:
fix white spaces

Andrey1079 committed:
Merge branch 'develop' into feature/p.7.6_add_project_to_favorite

Commit:
fix(projects cardprojects): add currentUser in import, del extra imports form CardProject

Julia-Papina committed:
style: style button more in profile volunteer

Commit:
fix(projects.jsx carpprojects.jsx): add userPorfile import to Projects, del exta imports in CardProj

Andrey1079 committed:
Merge pull request #185 from volunteers-for-city-projects/feature/p.7.6_add_project_to_favorite

p.7.6 add project to favorite#160

Julia-Papina committed:
feat: merge develop

KPVakhrushev committed:
feat(component): components for incomes

EvgeniiyaR committed:
Merge pull request #170 from volunteers-for-city-projects/fix/errors

Fix/errors

KPVakhrushev committed:
Merge branch 'develop' into feature/project-incomes

Julia-Papina committed:
feat: added style crumbs - route link, style card project and project like button

DBKai committed:
feat(project): added save project as draft

Julia-Papina committed:
Merge pull request #184 from volunteers-for-city-projects/feature/projects-personal-organization-api

Feature/projects personal organization api

DBKai committed:
fix(project): changed city for draft

DBKai committed:
Merge branch 'develop' into feature/project-draft

DBKai committed:
Merge pull request #187 from volunteers-for-city-projects/feature/project-draft

Feature/project draft

KPVakhrushev committed:
feat(component): modified income components, added component ModalContent

KPVakhrushev committed:
Merge branch 'develop' into feature/project-incomes

Julia-Papina committed:
feat: modal delete card project

n

DBKai committed:
feat(project): added initial values from localstorage and replaced custom input on daterange

DBKai committed:
feat(customdaterange): added styled-components for style clear button

Commit:
feat: add status of project in ProjectView

Commit:
Merge branch 'develop' into feat/p.7.3.1_view_cardProject

Commit:
feat(projectview): change styles for 1919px and less

change display project-view_container-name-place-date from flex to grid

Commit:
feat: add push message for projectStatus in ProjectView

fix #138

Commit:
fix: change styles for push message

Julia-Papina committed:
feat: profile volunteers project api, modal delete

Commit:
feat(cardproject): create ProjectEditButton

change card__status-btn on component ProjectEditButton

VanVorobyov committed:
feat(projects): add projects filter

VanVorobyov committed:
feat(projects): add projects filter

VanVorobyov committed:
feat(projects): add projects filter

Julia-Papina committed:
feat: card projects in profile volunteer, styles for page projects - grid container for selects

DBKai committed:
Merge pull request #189 from volunteers-for-city-projects/fix/project-draft

feat(project): added initial values from localstorage and replaced cu…

vvkon13 committed:
fix(login): change symmetry reset and restore password 1366px

n

vvkon13 committed:
fix(login): lf crlf

Commit:
feat: change view buttons in projectView for organaizer

change view buttons in projectView for organaizer, replace handlecardProject in CardProject

Commit:
fix: replace the values for ProjectView with var

Commit:
fix: change position of push-message at 1920px

Commit:
feat: add routes for btns and make Stub for apsent missing pages

Commit:
fix: fix clic buttons on cardProject

Commit:
fix: fix handleProject in CardProject

KPVakhrushev committed:
Merge pull request #188 from volunteers-for-city-projects/feature/project-incomes

Feature/project incomes

vvkon13 committed:
feat(showprojectstatus): add new project statuses

vvkon13 committed:
feat(cardproject): display delete edit buttons depending on the project status and the project owner

Julia-Papina committed:
Merge pull request #191 from volunteers-for-city-projects/feature/profile-volunteer-projects

feature/profile-volunteer-projects

Julia-Papina committed:
feat: modal succes

vvkon13 committed:
Merge branch 'develop' into feature/projects-preview-show-project-status

KPVakhrushev committed:
fix(incomes): fix errors in incomes components

Julia-Papina committed:
feat: test profile volunteer use effect

vvkon13 committed:
Merge pull request #194 from volunteers-for-city-projects/feature/projects-preview-show-project-status

Feature/projects preview show project status

vvkon13 committed:
feat(profilevolunteer): checking the connection

vvkon13 committed:
feat(profileorganization): implement the deletion of the project in the organizer's personal account

Commit:
fix: delete extra routes

Commit:
fix: hot fix eslint

KPVakhrushev committed:
fix(cardincome): fix image size

Commit:
Merge branch 'develop' into feat/LO4.6_view_cardProject_organizer

Commit:
fix: fix bugs

Andrey1079 committed:
Merge branch 'develop' into feat/p.7.3.1_view_cardProject

Andrey1079 committed:
Merge pull request #190 from volunteers-for-city-projects/feat/p.7.3.1_view_cardProject

Feat/p.7.3.1 view card project

KPVakhrushev committed:
fix(incomes): fix filter incomes by project and style income card

vvkon13 committed:
feat(cardproject): remove an extra component

vvkon13 committed:
feat: resolved merge conflict by incorporating both suggestions

VanVorobyov committed:
Merge branch 'develop' into feature/profile-filters

vvkon13 committed:
fix: lf crlf

vvkon13 committed:
Merge pull request #196 from volunteers-for-city-projects/feature/delete-card-project

Feature/delete card project

vvkon13 committed:
fix(carsproject): resolved merge conflict by incorporating both suggestions

vvkon13 committed:
Merge pull request #192 from volunteers-for-city-projects/fix/login-symmetry

Fix/login symmetry

VanVorobyov committed:
feat(projects): add project's filter

VanVorobyov committed:
fix(signup): remove delete buttons from selects and add required status to organize phonee

VanVorobyov committed:
fix(project): fix value of select option

VanVorobyov committed:
style(organize profile): add styles to filter's buttons

vvkon13 committed:
feat(projectlikebutton): add deletion of the project from personal account when disliking

Julia-Papina committed:
feat: page projects - styles 1366

vvkon13 committed:
fix(cardproject): delete console.log

vvkon13 committed:
Merge pull request #197 from volunteers-for-city-projects/feature/projects-card-like

feat(projectlikebutton): add deletion of the project from personal ac…

Julia-Papina committed:
Merge pull request #198 from volunteers-for-city-projects/feature/fix-styles-projects-page

Feature/fix styles projects page

KPVakhrushev committed:
fix(incomes): fix style, remove console, add style form 1366

KPVakhrushev committed:
Merge branch 'develop' into feature/project-incomes

KPVakhrushev committed:
Merge pull request #195 from volunteers-for-city-projects/feature/project-incomes

fix(incomes): fix errors in incomes components

KPVakhrushev committed:
Merge branch 'develop' into feat/LO4.6_view_cardProject_organizer

Commit:
fix: delet extra import in ProjectView

VanVorobyov committed:
Merge branch 'develop' into feature/profile-filters

# Conflicts:
#	src/components/CardIncome/CardIncome.scss
#	src/components/CardProject/CardProject.jsx

VanVorobyov committed:
fix: merge develop

VanVorobyov committed:
Merge pull request #199 from volunteers-for-city-projects/feature/profile-filters

Feature/profile filters

Julia-Papina committed:
fix: fix style card project in page projects (auto-fill)

Julia-Papina committed:
Merge pull request #200 from volunteers-for-city-projects/fix/style-projects

Fix/style projects

VanVorobyov committed:
fix(inputgroup): fix styles for selects in forms and projects page

VanVorobyov committed:
Merge pull request #201 from volunteers-for-city-projects/fix/forms

fix(inputgroup): fix styles for selects in forms and projects page

Julia-Papina committed:
fix: fix style projects cards on page projects on 1366 (auto-fill)

Julia-Papina committed:
Merge pull request #202 from volunteers-for-city-projects/fix/style-projects-1366

fix: fix style projects cards on page projects on 1366 (auto-fill)

KPVakhrushev committed:
fix(projectview): fix new income button visible

VanVorobyov committed:
feat(signup): add recaptcha to signup forms

Andrey1079 committed:
Merge branch 'develop' into feat/LO4.6_view_cardProject_organizer

Commit:
fix: fix

Andrey1079 committed:
Merge pull request #193 from volunteers-for-city-projects/feat/LO4.6_view_cardProject_organizer

Feat/lo4.6 view card project organizer

KPVakhrushev committed:
fix(cardincome): fix remove participant

KPVakhrushev committed:
Merge pull request #203 from volunteers-for-city-projects/fix/new-income-button

fix(projectview): fix new income button visible

KPVakhrushev committed:
refactor(projectview): project request moved to router loader

KPVakhrushev committed:
refactor(projectview): separate header and content in  ProjectView

Julia-Papina committed:
feat: mobile layout main page

Julia-Papina committed:
feat: header footer size mobile

VanVorobyov committed:
feat(organizeprofile): add filter to projexts

KPVakhrushev committed:
refactor(pageproject): move crumbs to shared components Crumbs

KPVakhrushev committed:
Merge branch 'develop' into fix/incomes

KPVakhrushev committed:
fix(pageproject): remove crumbs dublicate on page

Julia-Papina committed:
feat: hamburger icon, profile icon, popup hamburger menu

KPVakhrushev committed:
fix(formincome): fix income form style and phone validation

KPVakhrushev committed:
fix(popup): fix popup  z-index

KPVakhrushev committed:
fix(formincome): fix phone format on server request

KPVakhrushev committed:
fix(modalcontent): fix text align

Julia-Papina committed:
feat: burger menu, buttons in burger menu

KPVakhrushev committed:
Merge pull request #204 from volunteers-for-city-projects/fix/incomes

Fix/incomes

VanVorobyov committed:
feat(profile): add filters for volunteer and organize profiles

VanVorobyov committed:
fix(minor fixes): minor fixes

VanVorobyov committed:
Merge branch 'develop' into feature/profile-filters

KPVakhrushev committed:
refactor(validation): separate fields from schema

Julia-Papina committed:
feat: changed styles components

DBKai committed:
fix(organizer): fixed create project response

Julia-Papina committed:
feat: added click on the overlay to close burger menu

Julia-Papina committed:
feat: page projects media screen max-width 450

Julia-Papina committed:
fix: fix error prop types in console

VanVorobyov committed:
Merge branch 'develop' into feature/profile-filters

VanVorobyov committed:
feat(profile-filter): add filters to prodiles, update dislike function

Julia-Papina committed:
Merge pull request #213 from volunteers-for-city-projects/feature/main-page-mobile

Feature/main page mobile

VanVorobyov committed:
Merge pull request #220 from volunteers-for-city-projects/feature/profile-filters

KPVakhrushev committed:
Merge pull request #212 from volunteers-for-city-projects/fix/incomes

refactor(validation): separate fields from schema

KPVakhrushev committed:
refactor(incomes): add server filter by projectID

KPVakhrushev committed:
fix(incomes): fix header margin

fix #219

KPVakhrushev committed:
fix(incomes): fix styles

KPVakhrushev committed:
Merge pull request #266 from volunteers-for-city-projects/fix/incomes

Fix/incomes

DBKai committed:
feat(router): added route to edit project

DBKai committed:
feat(cardproject): added button click handler for edit project

DBKai committed:
style(style): corrected white-spaces

DBKai committed:
feat(project): added loading project data into inputs

DBKai committed:
feat(customdaterange): added correct loading daterange

KPVakhrushev committed:
fix: fix button style issue 267

DBKai committed:
feat(project): added field validation when editing a project

KPVakhrushev committed:
fix: fix word break in cover letter

DBKai committed:
fix(project): added buttons margin

DBKai committed:
Merge pull request #268 from volunteers-for-city-projects/fix/incomes

fix: fix button style issue 267

DBKai committed:
fix(passwordrecovery): fixed form gap

DBKai committed:
fix(loginpasswordrecovery): fixed position component

Julia-Papina committed:
style: margin in styles profile

DBKai committed:
feat(preloader): added preloader component

DBKai committed:
feat(organizer): added edit project request

DBKai committed:
feat(app): added preloader on isLoading event

DBKai committed:
feat(project): added edit project and convert image from url to base64

Julia-Papina committed:
Merge pull request #272 from volunteers-for-city-projects/fix/styles-profile

style: margin in styles profile

DBKai committed:
feat(preloader): changed preloader color

DBKai committed:
feat(projectview): changed row gap

DBKai committed:
fix(volunteersignupform): fixed different letter case

DBKai committed:
style(styles): fix linter white-spaces

DBKai committed:
feat(request): changed base url

DBKai committed:
feat(index): added description and title

DBKai committed:
feat(favicon): added new favicon

DBKai committed:
style(styles): changed white-spaces and etc linter

DBKai committed:
Merge pull request #277 from volunteers-for-city-projects/fix/issue-271

Fix/issue 271

DBKai committed:
Merge pull request #276 from volunteers-for-city-projects/fix/issue-273

feat(projectview): changed row gap

DBKai committed:
Merge pull request #274 from volunteers-for-city-projects/feature/edit-project

Feature/edit project

DBKai committed:
feat(recaptcha): temporarily disabled recaptcha for registering

DBKai committed:
fix(error-message): moved error-message style to a separate file and changed font size

DBKai committed:
fix(project): changed min length events

DBKai committed:
fix(volunteersignupformschema): changed regular expression for email

DBKai committed:
feat(projectview): added gap for project view

DBKai committed:
feat(header): removed search input from header

DBKai committed:
feat(readme): updated readme

DBKai committed:
feat(readme): created links

DBKai committed:
fix(promo): fixed misprint

DBKai committed:
fix(cardproject): changed like condition

DBKai committed:
fix(router): changed crumbs for edit project route

DBKai committed:
fix(project): changed set current project from loader

DBKai committed:
Update README.md

Added presentation link

Julia-Papina committed:
fix style slider

Julia-Papina committed:
style(slider): fix styles

Julia-Papina committed:
Merge pull request #278 from volunteers-for-city-projects/fix/slider-style

Fix/slider style

Julia-Papina committed:
fix(request): change domain name

Julia-Papina committed:
Merge pull request #279 from volunteers-for-city-projects/fix/domain-name

fix(request): change domain name

Julia-Papina committed:
fix: changing domain, readme

Julia-Papina committed:
Merge pull request #280 from volunteers-for-city-projects/fix/domain

fix: changing domain, readme

Julia-Papina committed:
build(restarting the build): restart

Julia-Papina committed:
Merge pull request #281 from volunteers-for-city-projects/fix/deploy

build(restarting the build): restart

Julia-Papina committed:
refactor: readme

Julia-Papina committed:
Merge pull request #282 from volunteers-for-city-projects/fix/readme

refactor: readme

Julia-Papina committed:
readme, fix build

Julia-Papina committed:
Merge pull request #283 from volunteers-for-city-projects/fix/build

readme, fix build
